client/src/components/ReviewTile.js

```diff
@@ -12,11 +15,58 @@ const ReviewTile = (props) => {
     }
   })
 
+  const handleClick = async (event) => {
+    setShowVoteError(false)
+    let alreadyVoted = false
+    props.votes.forEach(vote => {
+      if(vote.userId === props.userId) {
+        alreadyVoted = true
+        setShowVoteError(true)
+      }
+    })
+    if(!alreadyVoted) {
+      const newVote = await addNewVoteToTable(props.userId, event.currentTarget.value, props.reviewId)
+      props.addNewVoteToPage(newVote, props.reviewId)
+    }
+  }
+  
+  let voteError
+  if(showVoteError) {
+    voteError = <p>You've already voted on this review!</p>
+  }
+
+  let buttonClass
+  if(props.userId) {
+    buttonClass = "vote_button"
+  } else {
+    buttonClass = "logged_out_button"
```

Comment:
I wouldn't recommend `display: none` to hide an element, but rather do not render the button if it should not appear on the page for the user to see. 

```
let button = null;
if (user) {
  button = <button>
}
```